src/main/java/net/neoforged/neoforge/common/world/chunk/TicketController.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.world.chunk;
+
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.ForcedChunksSavedData;
+
+/**
+ * A class used to manage chunk loading tickets associated with a specific ID.
+ * <p>
+ * Controllers must be registered via {@link RegisterTicketControllersEvent}. A controller that isn't registered will have all of its tickets discarded, and the chunks will no longer be forcefully loaded.
+ *
+ * @param id       the ID of this controller
+ * @param callback a callback to be called when the tickets are loaded, in order to validate whether they're still active or not. {@code null} should be used when a callback needn't be provided
+ */
+@ParametersAreNonnullByDefault
+public record TicketController(ResourceLocation id, @Nullable LoadingValidationCallback callback) {
```

Comment:
Do you want to validate that the resourcelocation is not null in the canonical constructor?

```java
{
    Objects.requireNonNull(id, "id must not be null");
}
```